### PR TITLE
Feature/implement error boundary hoc

### LINKED ISSUE #3263: Implement error handling HOC in each main react-component (pre-migration)
| Parent Issue | Blocked by |
|:---:| :---:|
| #3271 | #3313 |

**Description**
In order to handle JS exceptions from components, we need to implement the new component created in #3313 in each main react component. 
This should be useful if this issue is solved before the migration to React (then should be only one error boundary for the entire app).
 
 Tasks:
 - [x] Implement the HOC in: 
   - [x] Modules
   - [x] Agents
   - [x] Management
   - [x] Security
   - [x] Tools
   - [x] Settings

### PR DESCRIPTION
**Pull request by pablomarga**
Hi team this PR add the ErrorBoundary hoc to react components.

Closes #3263

### CODE REVIEW

File: public/components/agents/syscollector/main.tsx [hunk 1]
```diff
@@ -10,16 +10,12 @@
...
-export class MainSyscollector extends Component {
-
-  constructor(props) {
-    super(props);
-  }
-
-  render() {
-    return (<SyscollectorInventory agent={this.props.agent}></SyscollectorInventory> )
-  }
+function MainSyscollectorClass(props) {
```
Comment (gabiwassan) [inline public/components/agents/syscollector/main.tsx:17]: Could be better like arrow functions
`const MainSyscollectorClass = (props) => {`

File: public/components/common/modules/main.tsx [hunk 2]
```diff
@@ -27,9 +27,10 @@ import { getAngularModule, getDataPlugin, getUiSettings } from '../../../kibana-
 import { MainModuleAgent } from './main-agent'
 import { MainModuleOverview } from './main-overview';
 import store from '../../../redux/store';
-import WzReduxProvider from '../../../redux/wz-redux-provider.js';
+import { compose } from 'redux';
+import { withReduxProvider,withErrorBoundary } from '../hocs';
 
-export class MainModule extends Component {
+export const MainModule = compose (withErrorBoundary,withReduxProvider) (class MainModule extends Component {
```
Comment (gabiwassan) [inline public/components/common/modules/main.tsx:33]: Check prettier here

File: public/components/common/welcome/agents-welcome.js [hunk 3]
```diff
@@ -329,7 +327,7 @@ export class AgentsWelcome extends Component {
             </EuiFlexGroup>
           </EuiFlexItem>
         </EuiFlexGroup>
-    );
+    );withErrorBoundary
```
Comment (gabiwassan) [inline public/components/common/welcome/agents-welcome.js:330]: Maybe type error?

File: public/components/management/cluster/cluster-visualization.js [hunk 4]
```diff
@@ -1,23 +1,14 @@
...
-      <div style={{ height: '100%' }}>
-        <WzReduxProvider>
-          <KibanaVis
-            visID={this.props.visId}
-            tab={this.props.tab}></KibanaVis>
-        </WzReduxProvider>
-      </div>
-    );
-  }
+function KibanaVisClass(props) {
```
Comment (gabiwassan) [inline public/components/management/cluster/cluster-visualization.js:6]: Same here change for arrow functions

**Review by gabiwassan — CHANGES_REQUESTED** [review]
Great job!
Just some comments. 
And check to apply prettier on all your changes. 💪🏼

**Review by frankeros — APPROVED** [review]
LGTM!

**Review by Machi3mfl — APPROVED** [review]
LGTM!

**Review by gabiwassan — APPROVED** [review]
LGTM ! 🚀

**Pull request merged**